clients/src/main/java/org/apache/kafka/clients/consumer/RangeAssignor.java

```diff
@@ -281,7 +281,7 @@ void onAssigned(String consumer, List<TopicPartition> newlyAssignedPartitions) {
             int numAssigned = numAssignedByConsumer.compute(consumer, (c, n) -> n + newlyAssignedPartitions.size());
             if (numAssigned > numPartitionsPerConsumer)
                 remainingConsumersWithExtraPartition--;
-            unassignedPartitions.removeAll(newlyAssignedPartitions);
+            newlyAssignedPartitions.forEach(unassignedPartitions::remove);
```

Comment:
it can be faster when removing elements from a Set one by one, compared to using removeAll with a List

When using removeAll with a List and a Set, the method iterates through the entire List and, for each element, checks if it exists in the Set (which can be slower because List doesn't have efficient lookup). 

Using forEach with remove directly removes elements from the Set without the need for checking membership in the List repeatedly, which can be faster in some cases.